### PR TITLE
Definition

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

**Comment by AObuchow** [comment]
Tested this locally and it worked for submodule, parent (with relative path), and dependency. 
Will continue reviewing tomorrow but should be good to merge very soon (the conflicts should be fixed first however, if possible).

**Comment by mickaelistria** [comment]
I've rebased it, but didn't see a conflict in the code.

**Comment by AObuchow** [comment]
> I've rebased it, but didn't see a conflict in the code.

Seems ok with GitHub now :+1:

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	private File getArtifactLocation(Dependency dependency) {
+		File localArtifact = localRepositorySearcher.findLocalFile(dependency);
+		if (localArtifact != null) {
+			return localArtifact;
+		}
+		return null;
+	}
+
+	private LocationLink toLocation(File target, DOMElement element) {
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java]: Does this function need the `DOMElement element` parameter? It doesn't seem to use it.

Comment (mickaelistria) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java]: It should be used as 4th parameter of the LocationLink, as `originSelectionRange`.
I can fix it now or in a later commit, as you prefer.

Comment (AObuchow) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java]: Probably best to do it now as I'm still reviewing (if you have a moment) :)

Comment (mickaelistria) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java]: Done.

**Pull request closed without merging**